### ISSUE TITLE
Privacy setting to make Channel and Playlist private (and Instance Only) 

### ISSUE THREAD

**Issue by smilekison** [Chocobozzz/PeerTube#4733]
Hey there @Chocobozzz, I have two feature request at the moment. 

1) Options to make channel private or for Instance only. As, right now the channel is public. (Use case Scenario: What if i want to have a channel to share with my instance users only and in case of federation i want to remain isolated.)

2) I see that there is an option to share video to Public, Private, Unlisted and Internal. I was wondering if this feature will also be provided for Playlist as there is no Internal options available at the moment.

**Comment by Chocobozzz** [comment]
Hi,

Please create 1 issue per feature request :)

> Options to make channel private or for Instance only. As, right now the channel is public. (Use case Scenario: What if i want to have a channel to share with my instance users only and in case of federation i want to remain isolated.)

I'm not sure to understand: what's the difference with channel's videos set as *internal*? Why do you need to hide the channel?

**Comment by smilekison** [comment]
Hi,
As an example, Let's say I am running PeerTube in an organization. And i have partner companies where i have used PeerTube as a federation to share resources. Now, with Federation, all the channels and videos are shared between different instances. Now, it will be easier if i can create a channel to share inside my instance only (Just like video Internal Setting). In this way, i can share channel with inside users only. I hope you got the idea.

**Comment by Chocobozzz** [comment]
Then if the channel only have "internal" videos it should not be federated

**Comment by smilekison** [comment]
Yes, I was thinking the channel and playlist privacy to internal will be an added feature for PeerTube Federation.

**Comment by Casuallynoted** [comment]
I definitely think having playlists be able to be set to "instance users only"/internal would be super helpful. Atm I have an instance that has some videos on an unlisted playlist, which is WIP stuff I really don't want to be federated but I'd love to have available to the users of my own instance. Would be super useful vs having to manually share the link.

**Comment by TangentFoxy** [comment]
I have been wanting this feature for *years*. I use an instance almost entirely internally because no other platform handles videos well enough for me, but I really hate showing a bunch of playlists to the world that they can't access because they're for internal videos only. It's also a bit of a privacy leak, and while I could put the whole instance behind a separate auth layer to protect against this, it's not quite worth that hassle.

**Comment by candidexmedia** [comment]
> I'm not sure to understand: what's the difference with channel's videos set as internal? Why do you need to hide the channel?

Because otherwise, you get a bunch of empty channels (publicly) that people are subscribing to. Since we cannot have videos that don't go in a channel, why not also have private channels?

> It's also a bit of a privacy leak, and while I could put the whole instance behind a separate auth layer to protect against this, it's not quite worth that hassle.

Excellent point re: privacy leak
